src/test/java/pl/onsight/wypozyczalnia/service/ProductTest.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @Test
+    @Transactional
+    public void shouldFindMoreThan1FromOriginalSizeOfProducts() {
+        //given
+        ProductEntity product = new ProductEntity();
+        productService.addProduct(product);
+
+        //when
+        List<ProductEntity> products = productService.findAllProducts();
+
+        //then
+        assertThat(products.size()).isEqualTo(sizeOfDatabase + 1);
+    }
+
+    @Test
+    @Transactional
+    public void shouldCheckQuantityOf44() {
+        //given
+        ProductEntity product = new ProductEntity();
+        product.setProductName("Karabinek");
+        int quantity = 44;
+        product.setQuantity(quantity);
+        productService.addProduct(product);
+
+        //when
+        List<CountProducts> countProducts = productService.countAllProductsByName();
+
+        //then
+        assertThat(countProducts.get(countProducts.size() - 1).getCount()).isEqualTo(quantity);
+    }
+
+    @Test
+    @Transactional
+    public void shouldFindProduct() {
+        //given
+        ProductEntity product = new ProductEntity();
+        product.setProductName("Buty");
+        productService.addProduct(product);
+
+        //when
+        List<CountProducts> countProducts = productService.countAllProductsByNameFiltered(product.getProductName());
+
+        //then
+        assertThat(countProducts.get(0).getProduct().getProductName()).isEqualTo(product.getProductName());
+    }
+
+    @Test
+    @Transactional
+    public void shouldFind2Products() {
+        //given
+        ProductEntity product = new ProductEntity();
+        product.setProductName("Buty");
+        productService.addProduct(product);
+
+        ProductEntity product2 = new ProductEntity();
+        product2.setProductName("Kask");
+        productService.addProduct(product2);
+
+        List<ProductEntity> productList = new LinkedList<>();
+        productList.add(product);
+        productList.add(product);
+        productList.add(product2);
+
+        //when
+        List<CountProducts> countProducts = productService.countProductsInProductList(productList);
+
+        //then
+        assertThat(countProducts.size()).isEqualTo(2);
+    }
+
+    @Test
+    @Transactional
+    public void shouldFound2Links() {
+        //given
+        ProductEntity product = new ProductEntity();
+        product.setProductName("snieg");
+        product.setTags("woda,ogien,zima");
+        productService.addProduct(product);
+
+        ProductEntity product2 = new ProductEntity();
+        product2.setProductName("deszcz");
+        product2.setTags("ogien,woda,komputer");
+        productService.addProduct(product2);
+
+        ProductEntity product3 = new ProductEntity();
+        product3.setProductName("grad");
+        product3.setTags("kaktus");
+        productService.addProduct(product3);
+
+        //when
+        Set<Link> setOfFoundLinks = productService.findRelatedProducts(productService.findProductById((long) (sizeOfDatabase + 2)));
```

Comment:
Tu nie powinno sizeOfDatabase + 0 albo +1    /  nie wskazuje teraz czasem na produkt 3 i wtedy related links powinien byc size 0. A tak to moze zmienić metodę addPrioduct z voida, zeby zwracała zapisany produkt... byłoby takie potwierdzenie ze jest sukces. i mozna dostać jego ID

Reply:
A tak to zajebiście

Reply:
myślałem nad tym, ale jeśli to miałoby zastosowanie tylko w testach to jest to moim zdaniem bez sensu. To się przydawało przy TOsach i mapperach, tutaj nie widzę takiego zastosowania. Chciałem wskazać na product2 i jako że id są od 1 to trzbea dodać 2.

Reply:
no kurde rzeczywiście.. liczyłem od 0 :)